File: app/src/main/java/com/shaheen/kbj/ImagesAdapter.java
package com.shaheen.kbj;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;


public class ImagesAdapter extends RecyclerView.Adapter<ImagesAdapter.ViewHolder>{
    private ArrayList<ImageModel> listdata;
    private Context context;

    // RecyclerView recyclerView;
    public ImagesAdapter(ArrayList<ImageModel> listdata, Context context) {
        this.listdata = listdata;
        this.context = context;
    }
    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        View listItem= layoutInflater.inflate(R.layout.item_image, parent, false);
        ViewHolder viewHolder = new ViewHolder(listItem);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        final ImageModel myListData = listdata.get(position);
        holder.imageView.setImageResource(myListData.getImage_id());
        holder.root.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                Intent intent = new Intent(context, JerseyEditActivity.class);
                intent.putExtra("item",myListData);
                context.startActivity(intent);


            }
        });
    }


    @Override
    public int getItemCount() {
        return listdata.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        public ImageView imageView;
        public CardView root;
        public ViewHolder(View itemView) {
            super(itemView);
            this.imageView = (ImageView) itemView.findViewById(R.id.image);
            this.root = (CardView) itemView.findViewById(R.id.root);
        }
    }
}

File: app/src/main/java/com/shaheen/kbj/adapter/ImagesAdapter.kt
package com.shaheen.kbj.adapter

import android.content.Context
import com.shaheen.kbj.model.ImageModel
import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import android.view.LayoutInflater
import com.shaheen.kbj.R
import android.content.Intent
import android.view.View
import android.widget.ImageView
import com.shaheen.kbj.activity.JerseyEditActivity
import androidx.cardview.widget.CardView
import java.util.ArrayList

class ImagesAdapter     // RecyclerView recyclerView;
    (private val listdata: ArrayList<ImageModel>, private val context: Context) :
    RecyclerView.Adapter<ImagesAdapter.ViewHolder>() {
    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val listItem =
            layoutInflater.inflate(R.layout.item_image, parent, false)
        return ViewHolder(listItem)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val myListData = listdata[position]
        holder.imageView.setImageResource(myListData.display_image_id)
        holder.root.setOnClickListener {
            val intent = Intent(context, JerseyEditActivity::class.java)
            intent.putExtra("item", myListData)
            context.startActivity(intent)
        }
    }

    override fun getItemCount(): Int {
        return listdata.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var imageView: ImageView
        var root: CardView

        init {
            imageView = itemView.findViewById<View>(R.id.image) as ImageView
            this.root = itemView.findViewById<View>(R.id.root) as CardView
        }
    }
}